File: app/src/main/java/com/deadcells/modding/AboutActivity.java
package com.deadcells.modding;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.Gravity;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

public class AboutActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        LinearLayout layout = new LinearLayout(this);
        layout.setOrientation(LinearLayout.VERTICAL);
        layout.setPadding(48, 96, 48, 48);
        layout.setGravity(Gravity.CENTER_HORIZONTAL);

        TextView title = new TextView(this);
        title.setText(R.string.about);
        title.setTextSize(24);
        title.setGravity(Gravity.CENTER);
        layout.addView(title);

        TextView space1 = new TextView(this);
        space1.setText("\n\n");
        layout.addView(space1);

        TextView dccm = new TextView(this);
        dccm.setText(getString(R.string.about_dccm_name));
        dccm.setTextSize(18);
        dccm.setGravity(Gravity.CENTER);
        layout.addView(dccm);

        TextView dccmDesc = new TextView(this);
        dccmDesc.setText(getString(R.string.about_dccm_desc));
        dccmDesc.setTextSize(14);
        dccmDesc.setGravity(Gravity.CENTER);
        layout.addView(dccmDesc);

        Button dccmBtn = new Button(this);
        dccmBtn.setText(getString(R.string.about_visit_dccm));
        dccmBtn.setOnClickListener(v -> startActivity(new Intent(Intent.ACTION_VIEW,
            Uri.parse("https://github.com/dead-cells-core-modding/core"))));
        layout.addView(dccmBtn);

        TextView space2 = new TextView(this);
        space2.setText("\n\n");
        layout.addView(space2);

        TextView alive = new TextView(this);
        alive.setText(getString(R.string.about_alive_name));
        alive.setTextSize(18);
        alive.setGravity(Gravity.CENTER);
        layout.addView(alive);

        TextView aliveDesc = new TextView(this);
        aliveDesc.setText(getString(R.string.about_alive_desc));
        aliveDesc.setTextSize(14);
        aliveDesc.setGravity(Gravity.CENTER);
        layout.addView(aliveDesc);

        Button aliveBtn = new Button(this);
        aliveBtn.setText(getString(R.string.about_visit_alive));
        aliveBtn.setOnClickListener(v -> startActivity(new Intent(Intent.ACTION_VIEW,
            Uri.parse("https://github.com/N3rdL0rd/alivecells"))));
        layout.addView(aliveBtn);

        setContentView(layout);
    }
}

File: app/src/main/java/com/deadcells/modding/AboutActivity.kt
package com.deadcells.modding

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.unit.dp
import com.deadcells.modding.ui.theme.DCMMTTheme
import top.yukonga.miuix.kmp.basic.Scaffold
import top.yukonga.miuix.kmp.basic.TopAppBar
import top.yukonga.miuix.kmp.preference.ArrowPreference

class AboutActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            DCMMTTheme {
                AboutScreen(onOpenUrl = { url ->
                    startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(url)))
                })
            }
        }
    }
}

@Composable
fun AboutScreen(onOpenUrl: (String) -> Unit) {
    Scaffold(
        topBar = {
            TopAppBar(title = stringResource(R.string.about))
        }
    ) { padding ->
        Column(
            Modifier
                .fillMaxSize()
                .padding(padding)
                .verticalScroll(rememberScrollState())
                .padding(12.dp)
        ) {
            ArrowPreference(
                title = stringResource(R.string.about_dccm_name),
                summary = stringResource(R.string.about_dccm_desc),
                onClick = { onOpenUrl("https://github.com/dead-cells-core-modding/core") }
            )

            ArrowPreference(
                title = stringResource(R.string.about_alive_name),
                summary = stringResource(R.string.about_alive_desc),
                onClick = { onOpenUrl("https://github.com/N3rdL0rd/alivecells") }
            )

            ArrowPreference(
                title = "Miuix",
                summary = "Compose Multiplatform UI library\nhttps://github.com/compose-miuix-ui/miuix\nLicensed under Apache-2.0",
                onClick = { onOpenUrl("https://github.com/compose-miuix-ui/miuix") }
            )
        }
    }
}
